exercises-fold/map/map.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Constructs and returns a new array with`fn` to each element of `collection`.
+ *
+ * In general,
+ *
+ * ```
+ * map([val1, val2, ..., valN], fn); // => [fn(val1), fn(val2), ..., fn(valN)]
+ * ```
+ *
+ * @example
+ * function double(x) {
+ *   return x * 2;
+ * }
+ *
+ * map([1, 2, 3, 4], double); // => [2, 4, 6, 8];
+ *
+ * @param {any[]} collection - An array containing anything
+ * @param {function} fn - A function that returns `true` or `false`
+ * @returns {any[]} A new array constructed by applying `fn` to each
+ *  element of `collection`
+ */
+function map(collection, fn) {
```

Comment:
```javascript
function map(collection, fn) {
  return fold(collection, function(acc, item) {
    acc.push(fn(item));
    return acc;
  }, []);
}
```

Reply:
In most languages, `push` will return the new array rather than the new length. Because JS doesn't it's hard to do this in one line. You could make a little helper that does this, like:

```javascript
function push(array, item) {
  array.push(item);
  return item;
}

function map(collection, fn) {
  return fold(collection, push, []);
}
```